Add pertpy

mschilli87 opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with a PR, please let the
right people know. There are language-specific teams for reviewing recipes.

| Language        | Name of review team           |
| --------------- | ----------------------------- |
| python          | `@conda-forge/help-python`    |
| python/c hybrid | `@conda-forge/help-python-c`  |
| r               | `@conda-forge/help-r`         |
| java            | `@conda-forge/help-java`      |
| nodejs          | `@conda-forge/help-nodejs`    |
| c/c++           | `@conda-forge/help-c-cpp`     |
| perl            | `@conda-forge/help-perl`      |
| Julia           | `@conda-forge/help-julia`     |
| ruby            | `@conda-forge/help-ruby`      |
| Rust            | `@conda-forge/help-rust`      |
| Go              | `@conda-forge/help-go`        |
| other           | `@conda-forge/staged-recipes` |

Once the PR is ready for review, please mention one of the teams above in a
new comment. i.e. `@conda-forge/help-some-language, ready for review!`
Then, a bot will label the PR as 'review-requested'.

Due to GitHub limitations, first time contributors to conda-forge are unable
to ping conda-forge teams directly, but you can [ask a bot to ping the team][1]
using a special command in a comment on the PR to get the attention of the
`staged-recipes` team. You can also consider asking on our [Zulip chat][2]
if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://conda-forge.zulipchat.com

All apologies in advance if your recipe PR does not receive prompt attention.
This is a high volume repository and the reviewers are volunteers. Review times
vary depending on the number of reviewers on a given language team and may be days
or weeks. We are always looking for more staged-recipe reviewers. If you are
interested in volunteering, please contact a member of @conda-forge/core.
We'd love to have the help!
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

---
I am willing to maintain the package and thus be listed in the maintainer section.

---
Depends on https://github.com/conda-forge/staged-recipes/pull/29955.
Supersedes https://github.com/conda-forge/staged-recipes/pull/29655.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/pertpy-coda/meta.yaml```) and found it was in an excellent condition.

I do have some suggestions for making it better though...


For **recipes/pertpy-coda/meta.yaml**:

 * ℹ️ `noarch: python` recipes should usually follow the syntax in our [documentation](https://conda-forge.org/docs/maintainer/knowledge_base/#noarch-python) for specifying the Python version.
   - For the `test.requires` section of the recipe, you should usually use the pin `python {{ python_min }}` for the `python` entry.
   - If the package requires a newer Python version than the currently supported minimum version on `conda-forge`, you can override the `python_min` variable by adding a Jinja2 `set` statement at the top of your recipe (or using an equivalent `context` variable for v1 recipes).


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/14892419516](https://github.com/conda-forge/conda-forge-webservices/actions/runs/14892419516). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/pertpy/meta.yaml```) and found it was in an excellent condition.

mschilli87 commented:
@Zethon: Please confirm that you are willing to co-maintain this package by commenting on this PR.

Zethson commented:
I am willing to co-maintain this recipe.

Thank you!

Zethson commented on recipes/pertpy/meta.yaml:
```diff
@@ -0,0 +1,63 @@
+
+about:
+  home: https://pertpy.readthedocs.io
+  summary: Perturbation Analysis in the scverse ecosystem (incl. [ta]scCODA support).
```
```suggestion
  summary: Perturbation Analysis in the scverse ecosystem
```

Nit: A bit noisy so I'd just omit this detail.

mschilli87 replied:
Absolutely. Plus, it's a lie. tascCODA is actually not included. This is a remnant of an earlier attempt to package an older version.
Thank you for the catch!

I'll do this locally, squash and force-push once I want to re-run the checks.

mschilli87 commented:
@Zethson: Thx!

fyi: The build failures are due to `fast-array-utils` not (yet) being available. I already created the feedstock and it should trickle down within the next hour or so and then I'll re-trigger the checks here.
So sorry in advance for the notifcation noise. 😉

mschilli87 commented:
Stupid typo. My bad. Once more. Sorry.

mschilli87 commented:
It's clearly getting late - bumped the version without updating the checksum. 🤦🏻 
Let's hope this is the last round.

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/pertpy/meta.yaml```) and found some lint.

Here's what I've got...


For **recipes/pertpy/meta.yaml**:

 * ❌ Selectors are suggested to take a ``<two spaces>#<one space>[<expression>]`` form. See lines [13]
 * ❌ `noarch` packages can't have skips with selectors. If the selectors are necessary, please remove `noarch: python`, or selector on line 14:
  skip: true # [win]



<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/15215195519](https://github.com/conda-forge/conda-forge-webservices/actions/runs/15215195519). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/pertpy/meta.yaml```) and found it was in an excellent condition.

mschilli87 commented:
Squashed and rebased, Windows build will probably fail (again) but Microsoft is not paying me so I won't fix their propriatory mess. 🤷🏻
Linux (and Mac) should be fine now. 🎉

mschilli87 commented:
@conda-forge/help-python: I consider this is ready for review. 🚀
